src/parser/statement.js

```diff
@@ -491,12 +491,14 @@ export default class StatementParser extends ExpressionParser {
     if (this.match(tt._catch)) {
       const clause = this.startNode();
       this.next();
-
-      this.expect(tt.parenL);
-      clause.param = this.parseBindingAtom();
-      this.checkLVal(clause.param, true, Object.create(null), "catch clause");
-      this.expect(tt.parenR);
-
+      if (this.match(tt.parenL) || !this.hasPlugin("optionalCatchBinding")) {
```

Comment:
Ok never mind, talked to Kara and it makes sense. This is for the case you have the plugin on, but you choose to still use the binding itself.